charts/ckan/templates/ingress.yaml

```
@@ -0,0 +1,29 @@
+{{- if .Values.ckan.ingress.enabled -}}
+{{- $fullName := include "common.names.fullname" . -}}
+{{- $svcName := printf "%s-%s" (include "common.names.fullname" .) "ckan"  -}}
+apiVersion: {{ include "common.capabilities.ingress.apiVersion" . }}
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    {{- include "common.labels.standard" . | nindent 4 }}
+  {{- with .Values.ckan.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.ckan.ingress.className }}
+  ingressClassName: {{ .Values.ckan.ingress.className }}
+  {{- end }}
+  tls:
+    - hosts:
+        - {{ required "You must provide a host, when enabling ingress" .Values.ckan.ingress.baseDomain | quote }}
+      secretName: {{ include "common.secrets.name" (dict "existingSecret" .Values.ckan.ingress.existingCertificate "defaultNameSuffix" "custom-certificate" "context" $) }}
```

The `defaultNameSuffix` is used when no existingCertificate is given, therefore it should be something like `certificate`

```suggestion
      secretName: {{ include "common.secrets.name" (dict "existingSecret" .Values.ckan.ingress.existingCertificate "defaultNameSuffix" "certificate" "context" $) }}
```